librepo/yum.c

```diff
@@ -152,9 +161,23 @@ lr_yum_repomd_record_enabled(LrHandle *handle, const char *type)
                 return TRUE;
             x++;
         }
+        // Substitution check
+        if (handle->yumslist) {
+            for (GSList *elem = handle->yumslist; elem; elem = g_slist_next(elem)) {
+                LrVar* subs = elem->data;
+                if (!g_strcmp0(subs->val, type)) {
+                    char *orig = subs->var;
+                    for (guint i = 0; handle->yumdlist[i]; i++) {
```

Comment:
I see that you iterate over ``yumdlist`` and not over ``yumslist`` which is guaranteed to be not NULL in this part of code. For that reason (yumdlist may be NULL here), there should be:

    for (guint i = 0; handle->yumdlist && handle->yumdlist[i]; i++) {

I.e. make sure that ``handle->yumdlist`` is not NULL before to try access its elements via index